src/modules/Sidebar/Sidebar.js

```diff
@@ -70,10 +77,40 @@ class Sidebar extends Component {
     this.stopAnimatingTimer = setTimeout(() => this.setState({ animating: false }), duration)
   }
 
+  componentDidMount() {
+    if (this.props.visible && this.props.closable) {
+      document.addEventListener('click', this.handleDocumentClick)
+    }
+  }
+
   componentWillReceiveProps(nextProps) {
     if (nextProps.visible !== this.props.visible) {
       this.startAnimating()
+      if (nextProps.visible && this.props.closable) {
+        document.addEventListener('click', this.handleDocumentClick)
+      } else {
+        document.removeEventListener('click', this.handleDocumentClick)
+      }
+    }
+  }
+
+  handleDocumentClick = (e) => {
+    const sidebarDOMNode = ReactDOM.findDOMNode(this.ref)
+    // close Sidebar if e.target is outside of Sidebar
+    if (!this.checkIfSidebarContainsClickTarget(e.target, sidebarDOMNode)) {
+      this.props.toggleVisibility()
```

Comment:
It looks like the Sidebar is already extending AutoControlledComponent, meaning it can `this.trySetState()` in an attempt to change it's state.  The state change will only be applied if there is no prop controlling the state keys.  Otherwise, it will delegate to the prop value.

In this case, `this.trySetState({ visible: false })` would attempt change the internal state, but the user's prop (if present) would win.  This allows us to rely only on `this.state.visible` internally and use `this.trySetState()` when we want to attempt to change the state.

The `toggleVisibility` prop should then be removed.